lib/add_to_wallet.dart

```diff
@@ -1,10 +1,39 @@
 import 'dart:async';
-import 'dart:io';
 
 import 'package:flutter/services.dart';
 
 export 'widgets/add_to_wallet_button.dart';
 
+
 class AddToWallet {
   static const MethodChannel _channel = const MethodChannel('add_to_wallet');
+
+  static final AddToWallet _instance = AddToWallet._internal();
+
+  /// Associate each Widget to its click event handlers
+  static final Map<String, FutureOr<dynamic> Function(MethodCall)> _handlers = Map();
+
+  factory AddToWallet() {
+    return _instance;
+  }
+
+  AddToWallet._internal() {
+    initMethodCallHandler();
+  }
+
+  void initMethodCallHandler() => _channel.setMethodCallHandler(_handleCalls);
+
+
+  Future<dynamic> _handleCalls(MethodCall call) async {
+    var handler = _handlers[call.arguments['key']];
+    return handler != null ? await handler(call) : null;
+  }
```

Comment:
Assuming the only message sent is that the button got pressed. If we had more events communication between the native and dart layer we will need to segregate over call.method, create an enum of events, etc...
Once again, at this level it's probably not worth it.